### PR TITLE
Move checkpoints to network (#1341)

### PR DESCRIPTION
**Pull request by Neurosploit**
This PR moves the checkpoints to the network file and adds methods to include them in the nodebuilder.
See https://github.com/stratisproject/StratisBitcoinFullNode/issues/1341

@dangershony please have a look. I'm still unsure where to move CheckpointInfo to.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus/Checkpoints.cs [hunk 1]
```diff
@@ -69,82 +45,17 @@ public CheckpointInfo(uint256 hash, uint256 stakeModifierV2 = null)
...
-        };
-
-        /// <summary>Checkpoints for the specific instance of the class and its network.</summary>
-        private readonly Dictionary<int, CheckpointInfo> networkSpecificCheckpoints;
+        /// <summary>The current network. </summary>
+        private Network network;
 
         /// <summary>Consensus settings for the full node.</summary>
-        private ConsensusSettings ConsensusSettings { get; }
+        private ConsensusSettings consensusSettings { get; }
```
Comment (mikedennis) [inline src/Stratis.Bitcoin.Features.Consensus/Checkpoints.cs]: I think this should be ConsensusSettings according to coding standard but confirming in slack.

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus/Checkpoints.cs]: Its private so change is correct, should start with lower case.

File: src/NBitcoin/Network.cs [hunk 2]
```diff
@@ -336,7 +363,8 @@ internal static Network Register(NetworkBuilder builder)
             network.DefaultPort = builder.Port;
             network.RPCPort = builder.RPCPort;
             network.genesis = builder.Genesis;
-            network.consensus.HashGenesisBlock = network.genesis.GetHash();
+            if(network.consensus != null && network.genesis != null)
```
Comment (dangershony) [inline src/NBitcoin/Network.cs]: This `if` is not correct both `consensus` and `genesis` should be set so we need to throw if they are not set.

**Comment by dangershony** [comment]
I have only one comment, very good change.

**Comment by Neurosploit** [comment]
@mikedennis I will move it to a seperate file. Dan didn't have any preferences.

File: src/NBitcoin/Networks.cs [hunk 3]
```diff
@@ -155,6 +155,32 @@ private static Network InitMain()
...
+                { 225430, new CheckpointInfo(new uint256("0x00000000000001c108384350f74090433e7fcf79a606b8e797f065b130575932")) },
+                { 250000, new CheckpointInfo(new uint256("0x000000000000003887df1f29024b06fc2200b55f8af8f35453d7be294df2d214")) },
+                { 279000, new CheckpointInfo(new uint256("0x0000000000000001ae8c72a0b0c301f67e3afca10e819efa9041e458e9bd7e40")) },
+                { 295000, new CheckpointInfo(new uint256("0x00000000000000004d9b4ef50f0f9d686fd69db2e03af35a100370c64632a983")) },
+                // Our own new checkpoints.
+                { 486000, new CheckpointInfo(new uint256("0x000000000000000000a2a8104d61651f76c666b70754d6e9346176385f7afa24")) },
+                { 491800, new CheckpointInfo(new uint256("0x000000000000000000d80de1f855902b50941bc3a3d0f71064d9613fd3943dc4")) },
+            };
+
+            foreach (var checkpoint in checkpoints)
```
Comment (noescape00) [inline src/NBitcoin/Networks.cs]: unvar please here and in the similar foreaches in this file

File: src/NBitcoin/Networks.cs [hunk 4]
```diff
@@ -155,6 +155,32 @@ private static Network InitMain()
...
+            {
+                network.checkpoints.Add(checkpoint.Key, checkpoint.Value);
```
Comment (noescape00) [inline src/NBitcoin/Networks.cs]: Maybe make Network.checkpoints non-readonly and just assign this dictionary to it (probably through a new method like SetCheckpoints or smth)?

Comment (Neurosploit) [reply src/NBitcoin/Networks.cs]: Instead I will follow the other lines in this file and remove the foreach completely.

Comment (noescape00) [reply src/NBitcoin/Networks.cs]: Nice!

**Comment by Aprogiena** [comment]
@noescape00 Have you tried to sync?

**Comment by noescape00** [comment]
No. But I doubt that there will be any problem with that. 
Can sync if you want.

**Comment by Aprogiena** [comment]
@noescape00 Yes please, always recheck all test plus basic syncing when we are doing reviews, at least up until a point when this is guaranteed to be checked automatically by our QA. Only Skip these tests in case of very small trivial PR's that cannot damage anything.

**Comment by noescape00** [comment]
Well, I was almost 100% that this PR will work as intended. After all- it's just a movement of checkpoints to the network. 

Ran the tests- testnet and mainnet synced successfully.

**Comment by Aprogiena** [comment]
then we can merge, tx

**Pull request merged**